#!/usr/bin/env python
PACKAGE='cuckoo_time_translator'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("switch_time",          double_t, 0,
        "Time after which switch to a pending clock filter and at the same time start a new pending filter (use 0 to disable switching).",
         0, 0.0, 36000)

algo_enum = gen.enum([ 
                       gen.const("None",       int_t, 0, "No device time. Receive time only is used."),
                       gen.const("ConvexHull", int_t, 1, "Convex hull"),
                       gen.const("Kalman",        int_t, 2, "Kalman filter"), 
                     ],
                     "Device timestamp filter algorithm")

gen.add("filter_algo", int_t, 0, "The filter algorithm used for device timestamps", 1, 0, 3, edit_method=algo_enum)

exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "DeviceTimeTranslator"))
